crates/swc_ecma_transforms_module/src/common_js.rs

```diff
@@ -89,6 +89,31 @@ where
         // Used only if export * exists
         let mut exported_names: Option<Ident> = None;
 
+        // make a preliminary pass through to collect exported names ahead of time
+        for item in items.clone() {
```

Comment:
I think we don't need `clone()`.

Reply:
I'm fairly new to Rust.  Is there a better way to iterate a vector twice?  Removing clone would cause the second iteration to fail.

Reply:
You can use `for item in &items` instead, as `&Vec<T>` implements `IntoIterator<Item = &T>`.

Reply:
Ah, thank you!  One of these days, I will learn Rust properly.